seqr/management/commands/check_bam_cram_paths.py

```diff
@@ -3,30 +3,49 @@
 from seqr.models import IgvSample
 from seqr.views.utils.dataset_utils import validate_alignment_dataset_path
 
+import collections
+import hail as hl
 import logging
+import tqdm
 
 logger = logging.getLogger(__name__)
 
+hl.init(log="/dev/null")
+
 
 class Command(BaseCommand):
+    help = 'Checks all gs:// bam or cram paths and, if a file no longer exists, deletes the path from the database'
 
     def add_arguments(self, parser):
-        parser.add_argument('args', nargs='*')
+        parser.add_argument(
+            '-d',
+            '--dry-run',
+            action="store_true",
+            help='Only print missing paths without updating the database',
+        )
+        parser.add_argument('args', nargs='*', help='only check paths in these project name(s)')
 
     def handle(self, *args, **options):
-        samples = IgvSample.objects.filter(
-            individual__family__project__name__in=args,
-        ).prefetch_related('individual', 'individual__family')
-
-        failed = []
-        for sample in samples:
-            try:
-                validate_alignment_dataset_path(sample.file_path)
-            except Exception as e:
+        samples = (IgvSample.objects.filter(
+            individual__family__project__name__in=args
+        ) if args else IgvSample.objects.all()).filter(
+            file_path__startswith='gs://'
+        ).prefetch_related('individual', 'individual__family__project')
+
+        missing_counter = collections.defaultdict(int)
+        for sample in tqdm.tqdm(samples, unit=" samples"):
+            if not hl.hadoop_is_file(sample.file_path):
                 individual_id = sample.individual.individual_id
-                failed.append(individual_id)
-                logger.info('Error at {} (Individual: {}): {} '.format(sample.file_path, individual_id, str(e)))
+                project = sample.individual.family.project.name
+                missing_counter[project] += 1
+                logger.info('Individual: {}  file not found: {}'.format(individual_id, sample.file_path))
+                if not options.get('dry_run'):
+                    sample.file_path = ""
+                    sample.save()
```

Comment:
so my only thought would be if we thought this was going to be a bunch of samples, you could save a list of the ids of the models that need to be updated and then do it as a bulk update at the end instead of potentially hundreds of database saves - `Sample.objects.filter(in__in=missing_sample_ids).update(file_path='')`

Reply:
good call